src/Spago/Git.purs

```diff
@@ -40,14 +41,19 @@ runGit args cwd = ExceptT do
     Right r -> Right r.stdout
     Left r -> Left r.stderr
 
-fetchRepo :: forall a b. { git :: String, ref :: String | a } -> FilePath -> Spago (GitEnv b) (Either (Array String) Unit)
+determineShouldCache :: forall b. String -> FilePath -> Spago (GitEnv b) Boolean
+determineShouldCache ref path = do
+  showRef <- Except.runExceptT $ runGit_ [ "show-ref", "--verify", "refs/heads/" <> ref ] (Just path)
+  pure $ isLeft showRef
```

Comment:
Now that the function is called `isBranch`, we need to return true if the command succeeds:
```suggestion
  pure $ isRight showRef
```